core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -73,7 +73,8 @@ class SparkEnv (
     val conf: SparkConf) extends Logging {
 
   @volatile private[spark] var isStopped = false
-  private val pythonWorkers = mutable.HashMap[(String, Map[String, String]), PythonWorkerFactory]()
+  private val pythonWorkers =
+    mutable.HashMap[(String, String, String, Map[String, String]), PythonWorkerFactory]()
```

Comment:
nit: shall we define a class to replace `(String, String, String, Map[String, String])`?